src/pages/AddServicePage.js

```diff
@@ -0,0 +1,101 @@
+/* @flow */
+
+import React from "react";
+import xhr from "xhr";
+
+import components from "../components";
+import sendEvent from "../google-tag-manager";
+
+declare var ISS_URL: string;
+
+/**
+ * Wraps the http request code in a promise.
+ *
+ * @param {Object} obj - data passed to http.request.
+ *
+ * @returns {Promise<Object>} a promise for the request.
+ */
+function _request(obj: XhrOptions) {
+    return new Promise((resolve, reject) => {
+        xhr(obj, (err, response, body) => {
+            if (response.statusCode == 200) {
+                resolve(response);
+            } else {
+                sendEvent({
+                    event: "xhr_failed",
+                    statusCode: response.statusCode,
+                });
+                reject(response);
+            }
+        })
+    });
+}
+
+class AddServicePage extends React.Component {
+
+    constructor(props): void {
+        super(props);
+
+        this.state = {
+            form: "Loading...",
+        };
+    }
+
+    async componentDidMount(): void {
+        let response = await _request({
```

Comment:
Perhaps. That was one of the options we considered, however the problem was that it won't inherit the styles of the page - e.g. custom fonts, etc.

I'll investigate an iframe version a bit more.